airflow/api_fastapi/core_api/routes/public/dag_parsing.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Annotated
+
+from fastapi import Depends, HTTPException, Request, status
+from fastapi.responses import Response
+from itsdangerous import BadSignature, URLSafeSerializer
+from sqlalchemy import exc, select
+from sqlalchemy.orm import Session
+
+from airflow.api_fastapi.common.db.common import get_session
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.auth.managers.models.resource_details import DagDetails
+from airflow.models.dag import DagModel
+from airflow.models.dagbag import DagPriorityParsingRequest
+
+if TYPE_CHECKING:
+    from airflow.auth.managers.models.batch_apis import IsAuthorizedDagRequest
+
+dag_parsing_router = AirflowRouter(tags=["DAG Parsing"], prefix="/parseDagFile/{file_token}")
+
+
+@dag_parsing_router.put(
+    "",
+    responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND]),
+)
+def reparse_dag_file(
+    file_token: str,
+    session: Annotated[Session, Depends(get_session)],
+    request: Request,
+) -> Response:
+    """Request re-parsing a DAG file."""
+    secret_key = request.app.state.secret_key
+    auth_s = URLSafeSerializer(secret_key)
+    try:
+        path = auth_s.loads(file_token)
+    except BadSignature:
+        raise HTTPException(status.HTTP_404_NOT_FOUND, "File not found")
+
+    requests: Sequence[IsAuthorizedDagRequest] = [
+        {"method": "PUT", "details": DagDetails(id=dag_id)}
+        for dag_id in session.scalars(select(DagModel.dag_id).where(DagModel.fileloc == path))
+    ]
+    if not requests:
+        raise HTTPException(status.HTTP_404_NOT_FOUND, "File not found")
+
+    parsing_request = DagPriorityParsingRequest(fileloc=path)
+    session.add(parsing_request)
+    try:
+        session.commit()
+    except exc.IntegrityError:
+        session.rollback()
+        return Response("Duplicate request", status_code=status.HTTP_201_CREATED)
```

Comment:
```suggestion
    session.commit()
```
we don't need to handle this.By default it will be raising a unique constraint error. Please refer to https://github.com/apache/airflow/pull/43875#discussion_r1856830840

Reply:
Indeed, you can remove this bit, this will be handled directly by the global exception handler.

Reply:
Modified the code to handle this